plugins/call_button/call_button.tsx

```diff
@@ -0,0 +1,139 @@
+import Menu from 'components/widgets/menu/menu';
+import {Constants} from 'utils/constants';
+
+import {Channel, ChannelMembership} from 'mattermost-redux/types/channels';
+import {PluginComponent} from 'types/store/plugins';
+
+import './call_button.scss';
+
+type Props = {
+    currentChannel: Channel;
+    channelMember?: ChannelMembership;
+    intl: IntlShape;
+    locale: string;
+    pluginCallComponents: PluginComponent[];
+    sidebarOpen: boolean;
+}
+
+function CallButton(props: Props) {
+    if (props.pluginCallComponents.length === 0) {
+        return null;
+    }
+
+    const {formatMessage} = props.intl;
+    const [active, setActive] = useState(false);
+    const [clickEnabled, setClickEnabled] = useState(true);
+    const prevSidebarOpen = useRef(props.sidebarOpen);
+
+    useEffect(() => {
+        if (prevSidebarOpen.current && !props.sidebarOpen) {
+            setClickEnabled(false);
+            setTimeout(() => {
+                setClickEnabled(true);
+            }, Constants.CHANNEL_HEADER_BUTTON_DISABLE_TIMEOUT);
+        }
+        prevSidebarOpen.current = props.sidebarOpen;
+    }, [props.sidebarOpen]);
+
+    const style = {
+        container: {
+            marginTop: '16px',
+            height: '32px',
+        } as CSSProperties,
+    };
+
+    if (props.pluginCallComponents.length === 1) {
+        const item = props.pluginCallComponents[0];
+
+        return (
+            <div
+                style={style.container}
+                className='flex-child'
+                onClick={() => {
+                    if (item.action && clickEnabled) {
+                        item.action(props.currentChannel, props.channelMember);
+                    }
+                }}
+                onTouchEnd={() => {
+                    if (item.action && clickEnabled) {
+                        item.action(props.currentChannel, props.channelMember);
+                    }
+                }}
+            >
+                {item.button}
+            </div>
+        );
+    }
+
+    const pluginCallComponents = props.pluginCallComponents.map((item) => {
+        return (
+            <li
+                className='MenuItem'
+                key={item.id}
+                onClick={(e) => {
+                    e.preventDefault();
+                    if (item.action) {
+                        item.action(props.currentChannel, props.channelMember);
+                    }
+                }}
+            >
+
+                {item.dropdownButton}
+            </li>
+        );
+    });
+    return (
+        <div
+            style={style.container}
+            className='flex-child'
+        >
+            <MenuWrapper
+                onToggle={(toggle: boolean) => {
+                    setActive(toggle);
+                }}
+            >
+                <button
+                    type='button'
+                    className={classNames('style--none call-button', 'dropdown', {active})}
+                >
```

Comment:
`button`s do by default have `type='button'`

```suggestion
                <button className={classNames('style--none call-button dropdown', {active})}>
```